tests/system/conftest.py

```diff
@@ -49,9 +49,9 @@ def skip_if_env_var_not_set(provider_env_vars):
 
 def pytest_collection_modifyitems(config, items):
     """Add @pytest.mark.system(provider_name) for every system test."""
-    rootdir = Path(config.rootdir)
+    rootdir = Path(config.rootpath)
```

Comment:
Do we still need to wrap the value in `Path()`? (Same for the line below.)

Reply:
I think Path is going to return a PosixPath object, which it was using before. Just a matter of where the input comes from.

This this change is blocking system tests from passing so I'm going to merge this, but if it turns out we can drop the `Path` we can circle back.

Reply:
For avoid guessing, what the return by specific attributes there is [`pytest.Config` API](https://docs.pytest.org/en/latest/reference/reference.html#pytest.Config)

Not a big deal to wrap `pathlib.Path` by subclass of `pathlib.Path`, time to time we even do it in Airflow codebase, so if someone want to fix it, as well as add type annotations into the hook input attributes, it could be done in follow up PR

Reply:
New PR #38691 with the requested changes.